libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/SharedApp.kt

```diff
@@ -54,11 +57,20 @@ object SharedApp {
         if (isSetup) return
         isSetup = true
 
-        MapboxNavigationApp.registerObserver(StateResetController(store))
+        MapboxNavigationApp.registerObserver(StateResetController(store, ignoreTripSessionUpdates))
         MapboxNavigationApp.lifecycleOwner.attachCreated(*navigationObservers)
         MapboxNavigationApp.registerObserver(audioGuidance ?: defaultAudioGuidance(context))
     }
 
+    fun tripSessionTransaction(updateSession: () -> Unit) {
```

Comment:
Should this be a `ThunkAction`? Or can it live in `StateResetController`?

Adding state logic to the SharedApp appears temporary considering all the logic is in delegated classes.

Reply:
What we need is a way to ignore any TripSession state changes when DropIn UI is responsible for modifying it. 

Not sure how `ThunkAction` would help here.

Reply:
>  can it live in `StateResetController`?

What do you mean by **it**? 
The `ignoreTripSessionUpdates` boolean flag, `fun tripSessionTransaction(updateSession: () -> Unit)` method or both?